core/src/lib.rs

```diff
@@ -363,4 +375,39 @@ mod tests {
             ),
         ]
     }
+
+    // tests the bucket method of Interval
+    #[test]
+    fn test_bucket() {
```

Comment:
- This function is misplaced, it should not be with the helper functions
- Rename it to something more descriptive. `interval_bucket` perhaps?